### PR TITLE
Support Andes ae350 ax45mp

### PR DESCRIPTION
**Pull request by lyctw**
This PR provides initial support for Andes AE350 platform (ax45mp 4 core) using mainline Linux kernel 5.18, U-Boot 2022.07 and OpenSBI v1.1. The generated wic image has been tested on AndeSight 5.2.0 bitmaps.

Key changes:
- ae350-ax45mp.conf: add machine configuration for Andes AE350 ax45mp 
- u-boot (v2022.07): add recipe and patches for AE350 ax45mp 
- linux (v5.18.14): add recipe for AE350 ax45mp, the source is hosted on Andes Github with mmc, dma, ethernet and rtc drivers support
- ae350-ax45mp-spl.wks: add partitioning command, the first partition holds u-boot-spl.bin (XIP mode SPL) and u-boot.itb can be burned to flash by using [SBI_Burn tool](https://github.com/andestech/Andes-Development-Kit), and also holds boot script, uEnv.txt and fitImage
- ae350-ax45mp.yml: add kas-container YAML for building poky distro

Changes since last [PR](https://github.com/riscv/meta-riscv/pull/352)
- docs: add ae350-ax45mp document
- add linux kernel SRCREV
- update upstream status of u-boot patches
- add u-boot fragments

### CODE REVIEW

File: ae350-ax45mp.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    BBPATH = "${TOPDIR}"
+    BBFILES ?= ""
+local_conf_header:
+  standard: |
+    CONF_VERSION = "2"
+    PACKAGE_CLASSES = "package_rpm"
+    SDKMACHINE = "x86_64"
+    # Use 'haveged' instead 'rng-tools' due to 'SIGSEGV' error during start 'rngd'
+    PACKAGE_EXCLUDE:append = "rng-tools"
+    IMAGE_INSTALL:append = "haveged"
```
Comment (kraj) [inline ae350-ax45mp.yml:43]: missing leading space after first \"

File: ae350-ax45mp.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    POKY_BBLAYERS_CONF_VERSION = "2"
+    BBPATH = "${TOPDIR}"
+    BBFILES ?= ""
+local_conf_header:
+  standard: |
+    CONF_VERSION = "2"
+    PACKAGE_CLASSES = "package_rpm"
+    SDKMACHINE = "x86_64"
+    # Use 'haveged' instead 'rng-tools' due to 'SIGSEGV' error during start 'rngd'
+    PACKAGE_EXCLUDE:append = "rng-tools"
```
Comment (kraj) [inline ae350-ax45mp.yml:42]: mssing leading space after \"

**Review by kraj — CHANGES_REQUESTED** [review]

File: ae350-ax45mp.yml (hunk 1 above)
Comment (lyctw) [reply ae350-ax45mp.yml]: Applied, thanks.

File: ae350-ax45mp.yml (hunk 2 above)
Comment (lyctw) [reply ae350-ax45mp.yml]: Applied, thanks.

**Review by kraj — APPROVED** [review]
LGTM now.

File: recipes-bsp/u-boot/files/ae350-ax45mp/0002-mmc-ftsdc010_mci-Support-DTS-of-ftsdc010-driver-for-.patch [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+The ftsdc010 driver has been implemented for generic dma in Linux
+kernel. And its compatible is andestech,atfsdc010g to distinguish
+the legacy andestech,atfsdc010 which is not for generic dma.
+
+Althought the ftsdc010_mci driver in U-Boot does not use dma, but
+it still can work well with the mmc node for generic dma. So add
+the compatible string to support it.
+
+Signed-off-by: Rick Chen <rick@andestech.com>
```
Comment (alistair23) [inline recipes-bsp/u-boot/files/ae350-ax45mp/0002-mmc-ftsdc010_mci-Support-DTS-of-ftsdc010-driver-for-.patch:15]: Can you add the upstream status?
https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines#Patch_Header_Recommendations:_Upstream-Status

**Review by alistair23 — COMMENTED** [review]
Looks good!

For future you don't need to close and create new PRs for every change. You can just push the changes to your original branch

File: recipes-bsp/u-boot/files/ae350-ax45mp/0002-mmc-ftsdc010_mci-Support-DTS-of-ftsdc010-driver-for-.patch (hunk 3 above)
Comment (lyctw) [reply recipes-bsp/u-boot/files/ae350-ax45mp/0002-mmc-ftsdc010_mci-Support-DTS-of-ftsdc010-driver-for-.patch:16]: I've added the status, thanks.

Comment (kraj) [reply recipes-bsp/u-boot/files/ae350-ax45mp/0002-mmc-ftsdc010_mci-Support-DTS-of-ftsdc010-driver-for-.patch:16]: seems good to go now, thanks for all work @lyctw

Comment (lyctw) [reply recipes-bsp/u-boot/files/ae350-ax45mp/0002-mmc-ftsdc010_mci-Support-DTS-of-ftsdc010-driver-for-.patch:16]: Thank @kraj and @alistair23 for helping me.

**Review by alistair23 — APPROVED** [review]

**Pull request merged**